k2/python/csrc/array.cc

```diff
@@ -100,11 +128,32 @@ class DLPackArray2<ValueType *, false, I> : public Array2<ValueType *, I> {
   std::unique_ptr<Tensor> indexes_tensor_;
   std::unique_ptr<Tensor> data_tensor_;
 };
-
 // Note: we can specialized for `StridedPtr` later if we need it.
 
 }  // namespace k2
 
+template <typename Ptr, typename I = int32_t>
+void PybindArray1Tpl(py::module &m, const char *name) {
+  using PyClass = k2::DLPackArray1<Ptr, I>;
+  using Parent = k2::Array1<Ptr, I>;
+  py::class_<PyClass, Parent>(m, name)
+      .def(py::init<py::capsule>(), py::arg("data"))
+      .def("empty", &PyClass::Empty)
+      .def(
+          "get_base",
+          [](PyClass &self) { return reinterpret_cast<Parent *>(&self); },
```

Comment:
```cpp
[](PyClass &self) -> Parent* {return self;}
```
and the explicit cast can be removed.

Reply:
I guess you mean `return &self`. Updated.